Added LeakCanary + Chucker + Network Config

Tgo1014 opened the pull request:
We used to have LeakCanary and Chucker but for some reason, they are gone. This PR adds them back as follows:
- `devDebug` builds: have LeakCanary so as devs are working on the app they can spot sooner when if something is leaking when they run the app
- `stagingDebug` builds: have Chucker, so QA can have access to network logs without requiring to have Charles
- `release`: should have none of these

I also added the network config file to support Charles in debug build. This should close #13.

johsoe commented:
I would rather enable real proxies instead of Chucker. 

QA keeps mentioning that our android network security config is not working and they cant inspect debug builds.

johsoe commented:
It would also be awesome if we added a basic setup for instrumentation tests detecting leaks

https://square.github.io/leakcanary/ui-tests/

johsoe commented:
> I would rather enable real proxies instead of Chucker.
> 
> QA keeps mentioning that our android network security config is not working and they cant inspect debug builds.

Could also be because we dont have a productionDebug build, now that I think of it

Tgo1014 commented:
Anwering your points @johsoe :

> I would rather enable real proxies instead of Chucker.

Do you want me to remove Chucker from this PR?
From what I see the template does not have the `networkSecurityConfig` file required for Charles to work. Maybe that's the reason it's not working for QAs. Should I add it to this PR for debug builds?

>It would also be awesome if we added a basic setup for instrumentation tests detecting leaks

The feature is very cool but it seems more like a way to confirm leaks are not happening after one is detected and fixed by writing the appropriate test to detect it's gone. I'm not sure we get much value just adding this into our template, especially because we don't have a CI template that runs instrumentation tests.

>Could also be because we dont have a productionDebug build, now that I think of it

I would expect that QAs are aware that they shouldn't be able to do it with production builds, just with staging.

levinzonr commented:
> From what I see the template does not have the networkSecurityConfig file required for Charles to work. Maybe that's the reason it's not working for QAs. Should I add it to this PR for debug 

I think Chuckker is great, even for developers, let's add the security config for the network so QAs can also play, it should be part of the template

> The feature is very cool but it seems more like a way to confirm leaks are not happening after one is detected and fixed by writing the appropriate test to detect it's gone. I'm not sure we get much value just adding this into our template, especially because we don't have a CI template that runs instrumentation tests.

Agreed, don't think its super beneficial as part of the template + there are no instrumentations tests in there

levinzonr commented:
Looks good to me :)
